db/db.go

```diff
@@ -124,11 +126,13 @@ func (db *Database) stateS3toDB(sf *statefile.File, path string, versionID strin
 	// Check if the associated lineage is already present in lineages table
 	// If so, it recovers its ID otherwise it inserts it at the same time as the state
 	var lineage types.Lineage
+	db.lock.Lock()
 	err = db.FirstOrCreate(&lineage, types.Lineage{Value: sf.Lineage}).Error
 	if err != nil || lineage.ID == 0 {
-		log.Error("Unknown error in stateS3toDB during lineage finding")
+		log.Errorf("Unknown error in stateS3toDB during lineage finding: %v", err)
```

Comment:
It could be interesting to use a structured log here (with fields).

Reply:
Done :+1: